inspectit-gepard-agent/src/test/java/rocks/inspectit/gepard/agent/integrationtest/spring/trace/TracingTest.java

```diff
@@ -52,69 +42,91 @@ void shouldNotSendSpansToBackendWhenNoScopesAreActive() throws Exception {
+    Optional<Span> childSpan = findSpan(spans, childSpanName);
+    boolean spansExist = parentSpan.isPresent() && childSpan.isPresent();
+
+    assertTrue(spansExist);
+
+    // We cannot compare the ByteStrings directly, because they are different objects
+    String parentId = parentSpan.get().getSpanId().toStringUtf8();
+    String childParentId = childSpan.get().getParentSpanId().toStringUtf8();
+    String parentTraceId = parentSpan.get().getTraceId().toStringUtf8();
+    String childTraceId = childSpan.get().getTraceId().toStringUtf8();
+
+    assertEquals(parentId, childParentId);
+    assertEquals(parentTraceId, childTraceId);
   }
 
   /**
-   * This method asserts that no spans with the given names exist.
+   * This method asserts that no spans from inspectIT with the given names exist.
    *
    * @param traces the collection of traces
    * @param parentSpanName the name of the parent span
    * @param childSpanName the name of the child span
    */
   private void assertNoSpans(
       Collection<ExportTraceServiceRequest> traces, String parentSpanName, String childSpanName) {
-    Stream<List<Span>> spanLists = getSpanLists(traces);
+    List<Span> spans = getInspectItSpans(traces);
+
+    Optional<Span> parentSpan = findSpan(spans, parentSpanName);
+    Optional<Span> childSpan = findSpan(spans, childSpanName);
+    boolean noSpanExist = parentSpan.isEmpty() && childSpan.isEmpty();
 
-    assertTrue(
-        spanLists.anyMatch(
-            spans -> {
-              Optional<Span> parentSpan = findSpan(spans, parentSpanName);
-              Optional<Span> childSpan = findSpan(spans, childSpanName);
+    assertTrue(noSpanExist);
+  }
 
-              return parentSpan.isEmpty() && childSpan.isEmpty();
-            }));
+  /**
+   * Maps the provided traces to a collection of spans.
+   *
+   * @param traces the collection of traces
+   * @return the collection of span within the provided traces
+   */
+  private List<Span> getSpans(Collection<ExportTraceServiceRequest> traces) {
+    return traces.stream()
+        .flatMap(
+            trace ->
+                trace.getResourceSpansList().stream()
+                    .flatMap(
+                        resourceSpans ->
+                            resourceSpans.getScopeSpansList().stream()
+                                .flatMap(scopeSpans -> scopeSpans.getSpansList().stream())))
+        .toList();
   }
 
   /**
-   * Maps the provided traces to a collection of span lists.
+   * Maps the provided traces to a collection of spans, which were created by inspectIT.
    *
    * @param traces the collection of traces
-   * @return the collection of span lists within the provided traces
+   * @return the collection of span within the provided traces
    */
-  private Stream<List<Span>> getSpanLists(Collection<ExportTraceServiceRequest> traces) {
+  private List<Span> getInspectItSpans(Collection<ExportTraceServiceRequest> traces) {
     return traces.stream()
```

Comment:
Could you try to make this a little more readable?
For my taste, this is a little to nested.